aws/resource_aws_prometheus_workspace.go

```diff
@@ -0,0 +1,111 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/prometheusservice"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsPrometheusWorkspace() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsPrometheusWorkspaceCreate,
+		Read:   resourceAwsPrometheusWorkspaceRead,
+		Update: resourceAwsPrometheusWorkspaceUpdate,
+		Delete: resourceAwsPrometheusWorkspaceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"alias": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"prometheus_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsPrometheusWorkspaceRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] Reading AMP workspace %s", d.Id())
+	conn := meta.(*AWSClient).prometheusconn
+
+	details, err := conn.DescribeWorkspace(&prometheusservice.DescribeWorkspaceInput{
+		WorkspaceId: aws.String(d.Id()),
+	})
+	if err != nil {
+		return err
```

Comment:
Errors returned from the AWS Go SDK should have message context for operators and code maintainers:

```suggestion
		return fmt.Errorf("error reading Prometheus Workspace (%s): %w", d.Id(), err)
```

Reply:
adopted